crates/runtime/src/traphandlers/unix.rs

```diff
@@ -155,6 +192,15 @@ unsafe fn get_pc(cx: *mut libc::c_void, _signum: libc::c_int) -> *const u8 {
             };
             let cx = &*(cx as *const libc::ucontext_t);
             (cx.uc_mcontext.psw.addr - trap_offset) as *const u8
+        } else if #[cfg(all(target_os = "macos", target_arch = "x86_64"))] {
```

Comment:
This code is taken from https://github.com/bytecodealliance/wasmtime/pull/2723/files#diff-560b66f5eac16eef20c78bcaf2c66582dbc6cf76fa1cc72faf6bca7b736cd660L205 without any changes.